common/include/common/common_utils.hpp

```diff
@@ -805,20 +711,13 @@ static inline std::vector<gbmv_param_t<scalar_t>> get_gbmv_params(Args& args) {
   if (args.csv_param.empty()) {
     warning_no_csv();
     std::vector<gbmv_param_t<scalar_t>> gbmv_default;
-    constexpr index_t dmin = 64, dmax = 1024;
-    constexpr index_t kmin = 1;
+    constexpr index_t dmin = 1024, dmax = 16384;
+    constexpr index_t kmin = 2048;
```

Comment:
I think that `kmin < dmin`, see
https://github.com/codeplaysoftware/sycl-blas/blob/6ec4f00e654570dc21aa00f61d711f46751db829/src/interface/blas2_interface.hpp#L478-L480

I suggest setting the minimum `k` as a function of the actual problem size, e.g., `m/32`

*Same applies for `tbmv` and `sbmv`.*